docs/design/gluster-block-provisioning.md

```diff
@@ -0,0 +1,350 @@
+* ```heketi``` looks for an appropriate gluster volume in one of the suitable
+  gluster clusters and calls out to ```gluster-block``` with the request to
+  create a block volume on the specified gluster file volume.
+* ```gluster-block``` creates a file on the gluster file volume and exports this
+  as an iSCSI target.
+* Upon success the resulting volume info (how to reach ...) is handed back
+  all the way to the provisioner.
+* The provisioner puts this volume information into a ```PersistentVolume```
+  (PV) and sets it up to use the iSCSI mount plugin for mounting.
+* The PV is bound to the original PVC.
+* The user references the PVC in an application description.
+* When the application pod is brought up on a node, the iSCSI mount plugin
+  takes care of initiating the block device, formatting (if required) and
+  mounting it.
+
+
+## Details About The ```glusterblock``` Provisioner
+
+The glusterblock provisioner is an external provisioner that is running in a
+container. It's code (currently WIP as of 2017-05-19) is located in the
+```gluster/block``` subdirectory of
+<https://github.com/kubernetes-incubator/external-storage>.
+
+The provisioner is mainly a simple translation engine that turns PVC requests
+into requests for heketi via heketi's RestFUL API, and wraps the resulting
+volume information into a PV.
+
+The provisioner is configured via a StorageClass which can look like this:
+
+```
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: glusterblock
+provisioner: gluster.org/glusterblock
+parameters:
+    resturl: "http://127.0.0.1:8081"
+    restuser: "admin"
+    secretnamespace: "default"
+    secretname: "heketi-secret"
+    hacount: "3"
+    clusterid: "630372ccdc720a92c681fb928f27b53f"
+```
+
+The available parameters are:
+
+* ```resturl```: how to reach heketi
+* ```restuser```, ```secretnamespace```, ```secretname```:
+  authentication information
+* ```hacount``` (optional): How many paths to the target server to configure.
+* ```clusterid``` (optional): List of one or more clusters to consider for
+  finding space for the requested volume.
+
+See  <https://github.com/kubernetes-incubator/external-storage/blob/master/gluster/block/README.md> for
+additional and up-to-date details.
+
+## Details About Heketi's New ```blockvolume``` Functionality
+
+For the purposes of gluster block volumes, the same heketi instance is used
+as for the regular glusterfs file volumes. Heketi has a new API though for
+treating blockvolume requests. Just as with the glusterfs file volume
+provisioning, the logic for finding suitable clusters and file volumes
+for hosting the loopback files is part of heketi.
+
+The API is a variation of the ```volume``` API and looks like this.
+
+The supported functions are:
+
+* ```BlockVolumeCreate```
+* ```BlockVolumeInfo```
+* ```BlockVolumeDelete```
+* ```BlockVolumeList```
+
+In the future, ```BlockVolumeExpand``` might get added.
```

Comment:
Yes, this makes sense, ExpandVolume or Resize is being targeted for Kubernetes 1.7, but the community has not landed on a design yet, so it makes sense to wait to add this operation